turbopack/crates/turbo-tasks/src/lib.rs

```diff
@@ -129,6 +129,8 @@ pub use vc::{
     VcValueTypeCast,
 };
 
+pub type SliceMap<K, V> = Box<[(K, V)]>;
```

Comment:
I think we should get rid of this type alias. Type aliases are useful when the reduce noise (e.g. the `FxHashMap` aliases to avoid the type parameter soup), but I don't think this does. `Box<[(K, V)]>` is an easy to write and understand type.

I *do* think we should have a newtype wrapper for `Box<[(K, V)]>` that enforces sorting and provides binary search lookups, but it seems like you don't need that for your use-case, so it's outside the scope of this PR.

Reply:
> I do think we should have a newtype wrapper for Box<[(K, V)]>

That is exactly why I added this new type (alias): so that we can replace it with a proper wrapper type once we need it somewhere.